internal/gps/constraints.go

```diff
@@ -53,8 +53,13 @@ type Constraint interface {
 	// design goal of the system.
 	typedString() string
 
-	// equals returns true if the constraints are logically equivalent.
-	equals(Constraint) bool
+	// identical returns true if the constraints are identical.
+	//
+	// Identical Constraints behave identically for all methods defined by the
+	// interface. A Constraint is always identical to itself.
+	//
+	// Constraints serialized for caching are de-serialized into identical instances.
```

Comment:
If the serialization process is stringifying, then this isn't true for branches, because of that default branch prop 😢